processor/common/common.go

```diff
@@ -0,0 +1,340 @@
+// Copyright 2019, OpenTelemetry Authors
```

Comment:
Can we move this to `internal/processor` package for now? I am reluctant to export all this at the moment, since it is will have to undergo breaking changes when moving from OC to OTLP in-memory representation).
Later, once we are certain we will no longer need to change the interfaces we can move to `process/common` if we want to.